src/Workspaces/Remote/ServiceHub/Services/SymbolFinder/RemoteSymbolFinderService.cs

```diff
@@ -227,19 +227,14 @@ public ValueTask OnDefinitionFoundAsync(SymbolGroup group, CancellationToken can
             }
 
             public async ValueTask OnReferencesFoundAsync(
-                IAsyncEnumerable<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references,
+                ImmutableArray<(SymbolGroup group, ISymbol symbol, ReferenceLocation location)> references,
                 CancellationToken cancellationToken)
             {
-                using var _ = ArrayBuilder<(SerializableSymbolGroup, SerializableSymbolAndProjectId, SerializableReferenceLocation)>.GetInstance(out var result);
-                await foreach (var (group, symbol, location) in references)
-                {
-                    result.Add((
-                        SerializableSymbolGroup.Dehydrate(_solution, group, cancellationToken),
-                         SerializableSymbolAndProjectId.Dehydrate(_solution, symbol, cancellationToken),
-                         SerializableReferenceLocation.Dehydrate(location, cancellationToken)));
-                }
+                var dehydrated = references.SelectAsArray(
+                    r => (SerializableSymbolGroup.Dehydrate(_solution, r.group, cancellationToken),
```

Comment:
>r  [](http://example.com/codeflow?start=20&length=2)

make static?

Reply:
Just noticed that the ArrayBuilder SelectAsArray methods should be using a FixedSizeBuilder. I can do that tomorrow , but it's sadly going to require compiler reviews.

Reply:
oh, woof, FixedSizeArrayBuilder isn't defined at the compiler level. never mind.